lib/Doctrine/DBAL/Connection.php

```diff
@@ -225,6 +209,8 @@ public function __construct(array $params, Driver $driver, Configuration $config
         $this->_expr = new Query\Expression\ExpressionBuilder($this);
 
         $this->autoCommit = $config->getAutoCommit();
+
+        $this->transactionManager = new TransactionManager($this);
```

Comment:
Separation of concerns, but technically they could very well be in the same class.

Reply:
It looks like they should be the same class because unlike queries created by the query builder or schemas managed by the schema manager, a transaction cannot exist outside of a connection, therefore the connection is a transaction manager.

Furthermore, currently, the concerns are only technically separated into separate classes. In fact, they mutually depend on each other (including instantiation of the transaction manager in the connection) which makes them a single whole.

Reply:
Makes sense.